I corrected errors and made an update to api-only.md (#4108)
jupyterhub/jupyterhub: docs/source/reference/api-only.md

> (api-only)=

Reviewer: This `(api-only)=` line is a [MyST target](https://myst-parser.readthedocs.io/en/latest/syntax/syntax.html#targets-and-cross-referencing), allowing other documents to link to this document by referencing `api-only` independent of what the title is etc.

It won't be visible when this document is built from a Markdown (.md) file into a HTML file using the tool Sphinx and MyST-parser.

Author: Thank you! @consideRatio...I'll update it in my next PR

Reviewer: Your'e welcome, thanks for your work!!

Note that you can update this PR by adding a commit. You could also replace the previous commit but that is good to avoid as then reviewers can loose track of what has changed since last inspection of a PR. When a commit is added as compared to replaced, its easy to see what has changed over time.

# Deploying JupyterHub In "API Only Mode"

Jupyterhub, as a service for deploying and managing jupiter servers for users, exposes its functionality, _primarily_ through a [REST API](rest).

For convenience, Jupyterhub also ships a basic web UI, built on the REST API. The basic web UI enables users to easily interact with their servers. E.g by clicking on the button to quickly start or stop their servers. It also enable admins to perform basic user and server mananagement tasks.

## The REST API

Previously, one could only achieve some admin UI functionalities through admin pages, such as paginated requests. The REST API provides additional functionality beyond what is available in the basic web UI. With Jupyterhub 2.0, the UI will always be composed from the REST API. This means that information on the pages will not be reliable, unless made available through the API.

## Why Rest API Only?

### Limited UI customization via templates

> The JupiterHub UI can be customized through extensivle HTML [templates](templates), but the scope at which it can be customized, is limited. It supports adding of content and messages to existing pages, but the page flow and available pages, cannot be customized.

Reviewer 2: Hi, I suggest you change the spelling of "JupiterHub" to "JupyterHub".

### Rich UI Customization with REST API based Apps

> JuptiterHub is primarily used as an API for managing Jupiter servers for other Jupiter-based appicatons that might want to present a different User Experience. Additionally, With an option of a fully customized experience available, you can now disable the hub UI and easily use your own pages, together with the JupiterHub Rest API, to build your own web applications to serve your users; relying on the hub only as an API for managing Users and Servers.

Reviewer 2: You should change the spelling here as well.

Author: Alright, noted! @chidimma

Reviewer 2: @Mackenzie is preferable, but you're welcome 👍🏾

(api-only)=

> # Deploying JupyterHub In "API Only Mode"

Reviewer 3: ```suggestion
# Deploying JupyterHub in "API only mode"
```

Jupyterhub, as a service for deploying and managing Jupyter servers for users, exposes its functionality, _primarily_ through a [REST API](rest).

For convenience, Jupyterhub also ships a basic web UI, built on the REST API. The basic web UI enables users to easily interact with their servers. E.g by clicking on the button to quickly start or stop their servers. It also enable admins to perform basic user and server mananagement tasks.

## The REST API

Previously, one could only achieve some admin UI functionalities through admin pages, such as paginated requests. The REST API provides additional functionality beyond what is available in the basic web UI. With Jupyterhub 2.0, the UI will always be composed from the REST API. This means that information on the pages will not be reliable, unless made available through the API.

> ## Why Rest API Only?

Reviewer 3: REST is an acronym, but perhaps this isn't quite the question we want to ask. How about:

```suggestion
## Why would I want to deploy JupyterHub with only its REST API?
```

### Limited UI customization via templates

> The JupyterHub UI can be customized through extensive HTML [templates](templates), but the scope at which it can be customized, is limited. It supports adding of content and messages to existing pages, but the page flow and available pages, cannot be customized.

Reviewer 3: ```suggestion
The JupyterHub UI can be customized through extensible HTML [templates](templates), but the scope at which it can be customized, is limited.
Custom templates support adding content and messages to existing pages, but the page flow and which pages are available cannot be customized.
```

### Rich UI Customization with REST API based Apps

> JupyterHub is primarily used as an API for managing Jupyter servers for other Jupyter-based applicatons that might want to present a different User Experience. However, With an option of a fully customized experience available, you can now disable the hub UI and easily use your own pages, together with the JupyterHub Rest API, to build your own web applications to serve your users; relying on the hub only as an API for managing Users and Servers.

Reviewer 3: This is no longer accurate. It is _not_ primarily used as an API. This is an increasingly common case, but not the primary case.

There are three values that are most logical for `hub_routespec`:

- `/` - This is the default, and used in most deployments. It is also the only option prior to JupyterHub 1.4.
- `/hub/` - This serves only Hub pages, both UI and API
- `/hub/api` - This serves _only the Hub API_, so all Hub UI is disabled. Aside from the OAuth confirmation page, if used.

If you choose a hub routespec other than `/`, the main JupyterHub feature you will lose is the automatic handling of requests for `/user/:username` when the requested server is not running.

> JupyterHub's handling of this request shows this page, telling you that the server is not running, with a button to launch it again:

Reviewer 3: Most of the changes here are to re-wrap text, which isn't an improvement. The lines are wrapped [on clauses or sentences](https://sembr.org).

We don't enforce that because I don't like to enforce any style that can't be autoformatted, but I'd prefer to avoid pure style changes that _remove_ the preferred style.
